diff --git a/node_modules/@uniswap/v3-sdk/dist/entities/tickDataProvider.d.ts b/node_modules/@uniswap/v3-sdk/dist/entities/tickDataProvider.d.ts
index ca9cefc..0f18fa8 100644
--- a/node_modules/@uniswap/v3-sdk/dist/entities/tickDataProvider.d.ts
+++ b/node_modules/@uniswap/v3-sdk/dist/entities/tickDataProvider.d.ts
@@ -1,4 +1,5 @@
 import { BigintIsh } from '@uniswap/sdk-core';
+import { Tick } from './tick';
 /**
  * Provides information about ticks
  */
@@ -7,9 +8,7 @@ export interface TickDataProvider {
      * Return information corresponding to a specific tick
      * @param tick the tick to load
      */
-    getTick(tick: number): Promise<{
-        liquidityNet: BigintIsh;
-    }>;
+    getTick(tick: number): Promise<Tick>;
     /**
      * Return the next tick that is initialized within a single word
      * @param tick The current tick
